posthog/hogql/test/test_resolver.py

```diff
@@ -694,3 +697,113 @@ def test_property_access_with_tuples_zero_index_error(self):
         ):
             node: ast.SelectQuery = self._select(query)
             resolve_types(node, context, dialect="clickhouse")
+
+    def test_resolve_field_through_linear_joins(self):
+        credential = DataWarehouseCredential.objects.create(
+            team=self.team, access_key="_accesskey", access_secret="_secret"
+        )
+
+        DataWarehouseTable.objects.create(
+            team=self.team,
+            name="subscriptions",
+            columns={
+                "id": "String",
+                "customer_id": "String",
+            },
+            credential=credential,
+            url_pattern="",
+        )
+
+        DataWarehouseTable.objects.create(
+            team=self.team,
+            name="customers",
+            columns={
+                "id": "String",
+                "email": "String",
+            },
+            credential=credential,
+            url_pattern="",
+        )
+
+        DataWarehouseJoin.objects.create(
+            team=self.team,
+            source_table_name="subscriptions",
+            source_table_key="customer_id",
+            joining_table_name="customers",
+            joining_table_key="id",
+            field_name="customer",
+        )
+
+        DataWarehouseJoin.objects.create(
+            team=self.team,
+            source_table_name="customers",
+            source_table_key="email",
+            joining_table_name="events",
+            joining_table_key="person.properties.$email",
+            field_name="events",
+        )
+
+        self.database = create_hogql_database(self.team.pk)
+        self.context = HogQLContext(database=self.database, team_id=self.team.pk, enable_select_queries=True)
+
+        node = self._select("""
+            SELECT
+                customer.events.distinct_id
+            FROM subscriptions
+        """)
+        node = cast(ast.SelectQuery, resolve_types(node, self.context, dialect="clickhouse"))
+
+    def test_resolve_field_through_nested_joins(self):
```

Comment:
@Gilbert09 I'd expect this test to fail but it passes for some reason. Do you see anything obviously wrong with it?

Reply:
Yep, that was it: https://github.com/PostHog/posthog/blob/22860343a03d3fe93f94b241ab485dee2870a07a/posthog/hogql/database/test/test_database.py#L876-L884 Thanks!

Reply:
Is this sufficient? print_ast won't run the query